src/composables/useUserDetails.ts

```diff
@@ -1,26 +1,36 @@
-import { shallowRef, watchEffect } from 'vue'
+import { createGlobalState, useAsyncState } from '@vueuse/core'
+import { watchEffect } from 'vue'
 import { Users } from '../lib/api/Users'
-import type { UserViewModel } from '../lib/api/data-contracts'
 import { useAuth } from './useAuth'
 import { useSessionToken } from './useSessionToken'
 
-export function useUserDetails() {
-	const userDetails = shallowRef<UserViewModel | null>(null)
+export const useUserDetails = createGlobalState(() => {
+	const users = new Users({
+		baseUrl: import.meta.env.VITE_BACKEND_URL,
```

Comment:
Might be worth making this a shared constant. I've seen this in a few files now

Reply:
It's mostly just a quirk of the codegen that we even have to instantiate a class here at all. It's basically just a wrapper for the api methods.